src/Endatix.Infrastructure/Setup/EndantixAppBuilderExtensions.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.ComponentModel;
+using Ardalis.GuardClauses;
+using Endatix.Core.Abstractions;
+using Endatix.Core.Infrastructure.Domain;
+using Endatix.Core.Infrastructure.Logging;
+using Endatix.Core.Infrastructure.Messaging;
+using Endatix.Core.Services;
+using Endatix.Framework.Hosting;
+using Endatix.Infrastructure;
+using Endatix.Infrastructure.Auth;
+using Endatix.Infrastructure.Data;
+using Endatix.Infrastructure.Email;
+using Endatix.Infrastructure.Setup;
+using FastEndpoints.Security;
+using MediatR;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Serilog;
+
+namespace Endatix;
+
+public static class EndatixHostBuilderExtensions
+{
+    public static IEndatixApp UseSerilogLogging(this IEndatixApp endatixApp)
+    {
+        endatixApp.WebHostBuilder.Host.UseSerilog((context, loggerConfig) =>
+                loggerConfig.ReadFrom.Configuration(context.Configuration));
+
+        endatixApp.LogBuilderInformation("Serilog logging configured");
+
+        return endatixApp;
+    }
+
+    public static IEndatixApp UseDomainServices(this IEndatixApp endatixApp)
+    {
+        endatixApp.Services.AddScoped<IFormService, FormService>();
+
+        return endatixApp;
+    }
+
+
+    public static IEndatixApp UseInfrastructure(this IEndatixApp endatixApp, Action<ConfigurationOptions> configuration)

```

Comment:
This method seems a bit long. I think it can easily have some of the logic extracted.

Having shorter methods helps for clarity and maintainability. I know that often it is not easy to find the balance how long a method is ok but aiming for shorter seems good approach to me. @PeterMarinov, what do you think?

Reply:
This is done as well. I agree with the comments. New `CreateSerilogLogger` method was added to extract it away for the already long and complex set up flow.